source/Halibut/Transport/Protocol/TemporaryFileStream.cs

```diff
@@ -20,6 +21,13 @@ public void SaveTo(string filePath)
 
             AttemptToDelete(filePath);
             File.Move(path, filePath);
+
+            //Update the permission on the file to match inherited permission of the new location.
```

Comment:
Instead of a comment, consider moving this code into a method named `SetFilePermissionsToInheritFromParent`.